bool power
int32 direction
int32 angle